### DISCUSSION TITLE
Can't get DIFF Output File (not HTML)

### DISCUSSION THREAD

**Discussion by asjones987** [WinMerge/winmerge#1356] [Q&A]
I was trying to generate a  TXT output file of differences (what are the changes in the new file vs the old file).
In looking at the official command line docs.
 https://manual.winmerge.org/en/Command_line.html

In further reading I see references to 
-cfg ReportFiles/ReportType=0   (or other number). 
-cfg ReportFiles/IncludeFileCmpReport=1 
-cfg Settings/DirViewExpandSubdirs=1

My impression is a few of these might do what I am after (with some tweaks). Can anyone provide guidance and info? 
I was doing a compare of text files
winmergeu /m Full /noninteractive   <old file path/name> <new file path/name>  /o c:\WinMerge-test\

Any guidance, thoughts, directions?

thanks

Alan

**Comment by sdottaka** [comment]
Unfortunately, WinMerge cannot output file comparison reports in any format other than HTML. (`-cfg ReportFiles/ReportType=0` is for folder comparison)
There is also no command line option for generating patches.

**Reply by asjones987** [reply]
this is very disappointing and surprising .... as the GUI has this under Tools -> Generate Patch.  
I had looked though the issues log but did not see a reference is this on the roadmap?

Are there any suggestions for a good OSS solution on Windows? I did some initial looking and all i found was older unsupported stuff.

thanks

Alan


note: I looked at 
http://gnuwin32.sourceforge.net/
http://unxutils.sourceforge.net/

**Reply by sdottaka** [reply]
> Are there any suggestions for a good OSS solution on Windows? I did some initial looking and all i found was older unsupported stuff.

Is there a problem with the Windows fc.exe command?

http://gnuwin32.sourceforge.net/packages/diffutils.htm is pretty old, but it also worked on Windows 11. (I tried unzipping and using the Binaries Zip file and the Dependencies Zip file)

**Reply by asjones987** [reply]
I did not realize the Windows fc.exe command was threre... the output in my test was not clear regarding differences (+/-) like in the normal package format.

really surprised there is not more out there for this
